x-pack/plugins/canvas/storybook/webpack.config.js

```diff
@@ -12,6 +12,10 @@ const { DLL_OUTPUT, KIBANA_ROOT } = require('./constants');
 
 // Extend the Storybook Webpack config with some customizations
 module.exports = async ({ config }) => {
+  const { module, plugins, resolve } = config;
+  const { rules } = module;
+  const { extensions, alias } = resolve;
```

Comment:
I would much prefer we use https://github.com/survivejs/webpack-merge rather than destructure the config like this. Having all these components of the parent config as simple variables like `plugins` and `alias` feels like it favors succinctness over understandability.

Reply:
If you're not a fan of `webpack-merge` I would compromise on using `baseConfig.module.rules` paths rather than reducing them to single words like `rules`.